qnexus/models/__init__.py

```diff
@@ -73,14 +76,19 @@ def df(self) -> pd.DataFrame:
 
 
 class Device(BaseModel):
-    """A device in Nexus, work-in-progress"""
+    """A device in Nexus."""
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
 
     backend_name: str
     device_name: Optional[str]
     nexus_hosted: bool
-    backend_info: BackendInfo
+    _backend_info: dict[str, Any]
 
-    model_config = ConfigDict(arbitrary_types_allowed=True)
+    @property
+    def backend_info(self) -> BackendInfo:
+        """The BackendInfo for a device."""
+        return BackendInfo.from_dict(self._backend_info)
```

Comment:
I've replaced the BackendInfo member attribute as Pydantic was getting unhappy when trying to generate the model schema (for some versions of pydantic/pydantic-core etc). This should preserve the desired behaviour and is simpler than the other alternatives I looked at working with the Pydantic internals